cpp/unit/attribute.h

```diff
@@ -84,6 +91,8 @@ class AttributeContainer {
 		type{t} {}
 
 	attr_type type;
+
+	virtual AttributeContainer *copy() const = 0;
```

Comment:
the container is mostly hidden, this was the only way I could get the templated get_attribute of units to work.